### DISCUSSION TITLE
Slash Command Varargs

### DISCUSSION THREAD

**Discussion by MaidScientistIzutsumiMarin** [tandemdude/hikari-lightbulb#162] [Q&A] [answered]
I may be missing something here, but I can't seem to figure out how to have slash commands get a variable number of arguments of a given type.

For example, if I want the user to be able to give any number of Users (greater than 0 preferably) in a slash command, how would I go about doing that?

**Comment by tandemdude** [comment, marked as answer]
Unfortunately this is not possible as discord does not support variable numbers of arguments. The closest you'd be able to achieve would be taking a string argument (comma or space delimited, etc) and parsing it into multiple arguments yourself.

**Reply by MaidScientistIzutsumiMarin** [reply]
That is what I ended up doing actually. I feel like this could be implemented without too much faff into lightbulb, which would definitely be nice. Or even if it just told Discord the slash command had the variable type as an optional argument the max amount of times... Or a suitable high number if there is no max.

The latter option would give the convenience of autocomplete for the users, even if it was just a duct tape solution for a duct tape API.

Regardless, thank you for the answer. I will mark this complete and possibly open a suggestion for the above feature.
